src/LocalCache.cc

```diff
@@ -556,34 +559,54 @@ bool LocalCachePrivate::FixPaths(const std::string &_modelVersionedDir)
 
 //////////////////////////////////////////////////
 void LocalCachePrivate::FixPathsInUri(tinyxml2::XMLElement *_elem,
-    const std::string &_modelVersionedDir)
+    const ModelIdentifier &_id)
 {
   if (!_elem)
     return;
 
-  std::string uri = _elem->GetText();
+  std::string oldUri = _elem->GetText();
   std::string prefix =  "model://";
 
   // Make sure the URI is of the form model://
-  if (uri.find(prefix) != std::string::npos)
-  {
-    int firstSlash = uri.find('/', prefix.size()+1);
-    std::string suffix = uri.substr(firstSlash);
+  if (oldUri.find(prefix) == std::string::npos)
+    return;
 
-    // Convert the model:// to point to an actual file.
-    // \todo(nkoenig) Handle URIs to other models. For
-    // example, ModelA may use something from ModelB.
-    std::string diskPath = common::joinPaths("file:/",
-        _modelVersionedDir, suffix);
+  int firstSlash = oldUri.find('/', prefix.size()+1);
 
-    _elem->SetText(diskPath.c_str());
+  auto resourceName = oldUri.substr(prefix.size(), firstSlash - prefix.size());
+
+  if (resourceName != _id.Name())
+  {
+    // On Blueprint and Citadel, just warn the user
+    // From Dome, use the name on the URI (resourceName) and assume the same
+    // owner
+    igndbg << "Model [" << _id.Name()
+           << "] loading resource from another model, named [" << resourceName
+           << "]. On Blueprint and Citadel, [" << resourceName << "] is "
+           << "ignored. From Dome, [" << _id.Name() << "] will be used. If ["
+           << resourceName << "] is not a model belonging to owner ["
+           << _id.Owner() << "], fix your SDF file!" << std::endl;
   }
+
+  auto filePath = oldUri.substr(firstSlash);
+
+  auto fuelUrl =
```

Comment:
Can you add a comment that describes what this is doing?

Reply:
05bf7412be60ce94bd612371ebfcaf7a2197ca20